integration-tests/src/test/java/com/example/IntegrationTest.java

```diff
@@ -158,6 +158,24 @@ public void optionalBindingNullable() {
     }
   }
 
+  @Test
+  @IgnoreCodegen
```

Comment:
This is actually compiling and working in codegen, but this test would fail at `fail()` with codegen.
I couldn't find any repro other than `OptionalBindingWrongScope` where linker was not null, any ideas?

Reply:
I merged the first two commits while I think about this more.

I didn't realize qualifier annotations on types actually worked...